snapshots/testsuite/testsuite.go

```diff
@@ -944,7 +957,7 @@ func check128LayersMount(name string) func(ctx context.Context, t *testing.T, sn
 				t.Fatalf("[layer %d] failed to apply on preparing dir: %+v", i, err)
 			}
 
-			if err := fstest.CheckDirectoryEqual(preparing, flat); err != nil {
+			if err := fstest.CheckDirectoryEqual(flat, preparing); err != nil {
```

Comment:
I'm inclined to keep them as is, assuming they are semantically same.

Reply:
@gabriel-samfira This patch is not small. So I think we should not touch un-related code. It can be handled in the followup.

Reply:
It is somewhat related. Without this, the test suite fails. The ```CheckDirectoryEqual()`` function needs the first folder to be the "parent" folder and the second one to be the "snapshot" of the parent. Reversing them breaks the comparison when ```metadataFiles``` are involved.

I can undo this, but then I'd also have to disable the tests, otherwise they will never pass.

Reply:
This change only touched three lines, according to https://github.com/containerd/containerd/pull/8043/commits/98236a851c001d00218121234c1106f213b80711.